contrib/zenith/libpagestore.c

```diff
@@ -50,10 +52,79 @@ page_server_api api = {
 static void
 zenith_connect()
 {
-	char	   *query;
-	int			ret;
+	char			 *query;
+	int				  ret;
+	char			 *auth_token;
+	char			 *err = NULL;
+	PQconninfoOption *conn_options;
+	PQconninfoOption *conn_option;
+	int 			 noptions = 0;
+
+    // this is heavily inspired by psql/command.c::do_connect
+	conn_options = PQconninfoParse(
+		page_server_connstring,
+	 	&err
+	);
+
+	if (conn_options == NULL) {
+		/* The error string is malloc'd, so we must free it explicitly */
+		char	   *errcopy = err ? pstrdup(err) : "out of memory";
+		PQfreemem(err);
+		ereport(ERROR,
+				(errcode(ERRCODE_SYNTAX_ERROR),
+					errmsg("invalid connection string syntax: %s", errcopy)));
+	}
+
+	auth_token = getenv(ZENITH_AUTH_TOKEN_ENV);
+	if (!auth_token) {
+		zenith_log(LOG, "No auth_token in env");
```

Comment:
Since current logic assumes that `page_server_connstring` should explicitly state that we want to use token (`password=ztoken`) I would rather move that check inside `if (strcmp(conn_option->val, "ztoken") == 0)` and throw an error if there is no corresponding env variable.